scikeras/utils/transformers.py

```diff
@@ -0,0 +1,210 @@
+from typing import Union
+
+import numpy as np
+import tensorflow as tf
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from sklearn.exceptions import NotFittedError
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import FunctionTransformer, OneHotEncoder, OrdinalEncoder
+from sklearn.utils.multiclass import type_of_target
+from tensorflow.keras.losses import Loss
+from tensorflow.python.keras.losses import is_categorical_crossentropy
+
+
+class TargetReshaper(BaseEstimator, TransformerMixin):
+    """Convert 1D targets to 2D and back.
+
+    For use in pipelines with transformers that only accept
+    2D inputs, like OneHotEncoder and OrdinalEncoder.
+
+    Attributes
+    ----------
+    ndim_ : int
+        Dimensions of `X` that the transformer was trained on.
+    """
+
+    def fit(self, X):
+        self.ndim_ = X.ndim
+        return self
+
+    @staticmethod
+    def transform(X):
+        if X.ndim == 1:
+            return X.reshape(-1, 1)
+        return X
+
+    def inverse_transform(self, X):
+        if not hasattr(self, "ndim_"):
+            raise NotFittedError(
+                f"This {self.__class__.__name__} is not initialized."
+                " You must call `fit` before using `inverse_transform`."
+            )
+        if self.ndim_ == 1 and X.ndim == 2:
+            return np.squeeze(X, axis=1)
+        return X
+
+
+class ClassifierLabelEncoder(BaseEstimator, TransformerMixin):
+    """Default target transformer for KerasClassifier.
+    """
+
+    def __init__(self, loss: Union[None, str, Loss] = None):
+        self.loss = loss
+
+    def fit(self, X: np.ndarray) -> "ClassifierLabelEncoder":
+        y = X  # rename for clarity, the input is always expected to be a target `y`
```

Comment:
Should we just rename the argument to `y`? That's what sklearn does for [LabelEncoder](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/preprocessing/_label.py#L227)

Reply:
Donzo